platform/wab/src/wab/client/components/grouping/VirtualTree.tsx

```diff
@@ -11,24 +12,25 @@ export interface RenderElementProps<T> {
     level: number;
     canOpen: boolean;
     isOpen: boolean;
+    isSelected?: boolean;
+    nodeAction?: NodeAction<T>;
     toggleExpand: () => void;
   };
 }
 
-export interface VirtualTreeRef {
-  expandAll: () => void;
-  collapseAll: () => void;
-}
-
-type Key = string;
+type NodeKey = string;
 
 type RenderElement<T> = React.FunctionComponent<RenderElementProps<T>>;
 
-interface LinearTreeNode<T> {
+type SelectDirection = 1 | -1;
+
+type NodeAction<T> = (node: T) => void | Promise<void>;
```

Comment:
I think the return should probably be `Promise<void>` only, so the user has to always handle it.